src/models/transfers.js

```diff
@@ -435,13 +437,18 @@ function * fulfillTransfer (transferId, fulfillmentUri) {
       }
     } catch (err) {
       // In case of a synchronization error, retry
-      if (Number(err.code) === 40001) {
-        retry(err)
+      if (Number(err.code) === 40001 || err.name === 'InsufficientFundsError') {
```

Comment:
Instead of always retrying whenever there is an `InsufficientFundsError`, should we better differentiate what the reason for the error was and only retry when the database transaction fails because it cannot be serialized?